examples/petsc/src/petscutils.c

```diff
@@ -136,6 +136,12 @@ PetscErrorCode SetupDMByDegree(DM dm, PetscInt degree, PetscInt num_comp_u,
                                &fe); CHKERRQ(ierr);
   ierr = DMSetFromOptions(dm); CHKERRQ(ierr);
   ierr = DMAddField(dm, NULL, (PetscObject)fe); CHKERRQ(ierr);
+  { /* force FE field, doesn't work on sphere */
```

Comment:
This works for our 3d problems, though there is a bug back in `DMPlexGetLocalOffsets()` now
```
Restriction offset 732 (-253701943) out of range [0, 1029]
```

Reply:
Ah, the issue is that the local size is completely wrong. I'm getting 1029 but it should be 192.